packages/gatsby/src/schema/schema.js

```diff
@@ -161,26 +171,79 @@ const addTypes = ({ schemaComposer, types, parentSpan }) => {
       })
 
       if (type) {
+        const typeName = type.getTypeName()
+        const createdFrom = `typeBuilder`
+        checkIsAllowedTypeName(typeName)
+        if (schemaComposer.has(typeName)) {
```

Comment:
Does this rely on order of added types?

I.e. if `default-site-plugin` adds type in `onPreBootstrap`, and some plugin adds types in `sourceNodes` - will we try to merge plugin type into type from site (which would fail)

Reply:
good point, yes, types added with the `createTypes` action are currently processed in the order they were added to the store.

the (documented, but not enforced) convention however is to only call `createTypes` from `sourceNodes` where `default-site-plugin` is guaranteed to run last (i think, actually have to check)

in the future, [the plan (which probably needs a bit more discussion)](https://github.com/gatsbyjs/gatsby/pull/13738#issuecomment-489318413) is to introduce one new Node API as the only place where schema customization stuff (createTypes, addThirdPartySchema, createFieldExtension) is allowed to be called. this would also solve the fact that the `createTypes` action currently can also be called in APIs running after schema generation, which means those will not be processed at all.

for now, we can either check the plugin name [right in the reducer](https://github.com/gatsbyjs/gatsby/blob/e47ed896723aa23979c9c7b1ecbede9312a8913d/packages/gatsby/src/redux/reducers/schema-customization.js#L21) to sort `default-site-plugin` to the end, or even simpler just sort [here](https://github.com/gatsbyjs/gatsby/blob/e47ed896723aa23979c9c7b1ecbede9312a8913d/packages/gatsby/src/schema/index.js#L15).